src/builder.cpp

```diff
@@ -509,10 +510,42 @@ MeshInstance3D* Builder::build_entity_mesh(int idx, LMEntity& ent, Node3D* paren
 	return mesh_instance;
 }
 
-String Builder::texture_path(const char* name)
+void Builder::load_and_cache_map_textures()
 {
-	//TODO: .png might not always be correct!
-	return String("res://textures/") + name + ".png";
+	m_loaded_map_textures.clear();
+
+	// Setup a texture extension list that both Trenchbroom and Godot support
+	constexpr int num_extensions = 9;
+	constexpr const char* supported_extensions[num_extensions] = { "png", "dds", "tga", "jpg", "jpeg", "bmp", "webp", "exr", "hdr" };
+	
+	// Attempt to load and cache textures used by the map
+	auto resource_loader = ResourceLoader::get_singleton();
+	const LMTextureData* tex;
+	String tex_path;
+
+	for (int tex_i = 0; tex_i < m_map->texture_count; tex_i++) {
+		bool has_loaded_texture(false);
```

Comment:
This might look nicer? 😊
```cpp
bool has_loaded_texture = false;
```

Reply:
Oops! That slipped through, will correct that.